### PR TITLE
Add manifest

### PR DESCRIPTION
**Pull request by staltz**
This should be the last PR I need merged before Manyverse can get ssb-db2, I think.

### CODE REVIEW

File: db.js [hunk 1]
```diff
@@ -21,12 +20,34 @@ function getId(msg) {
   return '%' + hash(JSON.stringify(msg, null, 2))
 }
 
-exports.init = function (sbot, dir, config) {
```
Comment (staltz) [inline db.js:24]: Used to be 3 args, now it's just 2, the `dir` arg is anyway just `config.path`.

File: db.js [hunk 2]
```diff
@@ -12,7 +12,6 @@ const { indexesPath } = require('./defaults')
 const Log = require('./log')
 const BaseIndex = require('./indexes/base')
 const Private = require('./indexes/private')
-const Migrate = require('./migrate')
```
Comment (staltz) [inline db.js:15]: The migrate plugin is no longer embedder inside db plugin. Instead, "ssb-db2" is db + migrate, see `index.js`.

File: db.js [hunk 3]
```diff
@@ -40,6 +61,12 @@ exports.init = function (sbot, dir, config) {
   const stateFeedsReady = DeferredPromise()
   let state = validate.initial()
 
+  sbot.close.hook(function (fn, args) {
+    close(() => {
+      fn.apply(this, args)
+    })
+  })
```
Comment (staltz) [inline db.js:68]: Moved from `index.js`

File: migrate.js [hunk 4]
```diff
@@ -98,30 +97,29 @@ exports.version = '0.4.0'
...
+        if (drainAborter) drainAborter.abort()
         fn.apply(this, args)
       })
       hasCloseHook = true
     }
     oldLogExists.set(fileExists(oldLogPath(config.path)))
     if (oldLogExists.value === false) {
       retryPeriod = Math.min(retryPeriod * 2, 8000)
-      retryTimer = setTimeout(fn, retryPeriod)
+      setTimeout(fn, retryPeriod).unref()
```
Comment (staltz) [inline migrate.js:122]: I should remember about `unref()`, it's in many cases better than keeping a ref to the timer (which forces Node.js to stay up)

File: test/base-index.js [hunk 5]
```diff
@@ -107,18 +107,6 @@ test('encrypted', (t) => {
 
     db.get(privateMsg.key, (err, msg) => {
       t.equal(msg.content.text, 'super secret')
-      t.end()
-    })
-  })
-})
-
-test('db.close', (t) => {
```
Comment (staltz) [inline test/base-index.js:115]: `db.close()` is no longer a separate API to be called, it's now automatically and internally done upon `sbot.close`. If you can think of a reason to close the db while keeping sbot up, let me know

File: db.js [hunk 6]
```diff
@@ -359,41 +386,27 @@ exports.init = function (sbot, dir, config) {
...
     publish,
+    query,
     del,
     deleteFeed,
     validateAndAdd,
     validateAndAddOOO,
     getStatus,
-    log,
-    close,
+    // close,
```
Comment (arj03) [inline db.js:399]: lets just remove this line

**Comment by arj03** [comment]
Looks good. Nitpicking like a pro with the close. Feel free to merge :-)

**Comment by staltz** [comment]
nitpick corrected! :)

**Pull request merged**